### PR TITLE
TypeActivate CodeGenCommand instead of new()

### PR DESCRIPTION
**Pull request by HaoK**
Removes GetRequiredService

Part of https://github.com/aspnet/Coherence-Signed/issues/186

@lodejard

### CODE REVIEW

File: src/Microsoft.Extensions.CodeGeneration/CodeGenCommand.cs [hunk 1]
```diff
@@ -5,42 +5,42 @@
...
-using System.Threading.Tasks;
-using Microsoft.Extensions.DependencyInjection;
 
 namespace Microsoft.Extensions.CodeGeneration
 {
     public class CodeGenCommand
     {
+        private readonly ILogger _logger;
+        private readonly ICodeGeneratorLocator _locator;
+
```
Comment (HaoK) [inline src/Microsoft.Extensions.CodeGeneration/CodeGenCommand.cs]: We probably could get rid of the IServiceProvider dependency now, but not sure if that's needed for consumer usage outside of this class

Comment (prafullbhosale) [inline src/Microsoft.Extensions.CodeGeneration/CodeGenCommand.cs]: Its not needed for consumer usage. 
Should be safe to remove. The ServiceProvider didn't need to be public in the first place.

Comment (HaoK) [inline src/Microsoft.Extensions.CodeGeneration/CodeGenCommand.cs]: Will do, does the change look good to you other than that?

Comment (prafullbhosale) [inline src/Microsoft.Extensions.CodeGeneration/CodeGenCommand.cs]: :+1:

**Comment by prafullbhosale** [comment]
:shipit:

**Comment by HaoK** [comment]
a2eaf1e5ec1c3d76fd0e15029535dcd28e74b18f

**Pull request closed without merging**